### ISSUE TITLE
astropy table with many long comments can't be written as a fits table.

### ISSUE THREAD

**Issue by segasai** [astropy/astropy#11627]
### Description
The writing of a table read from ascii into fits failed 

### Expected behavior
I read an ascii table and expect to be able to directly write it to fits

### Actual behavior
Exception 
```
Traceback (most recent call last):
  File "xx.py", line 6, in <module>
    tab.write('/tmp/xx.fits', overwrite=True)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/table/connect.py", line 127, in __call__
    registry.write(instance, *args, **kwargs)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/registry.py", line 563, in write
    writer(data, *args, **kwargs)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/connect.py", line 435, in write_table_fits
    table_hdu.writeto(output)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/hdu/base.py", line 1582, in writeto
    hdulist.writeto(name, output_verify, overwrite=overwrite,
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/utils/decorators.py", line 535, in wrapper
    return function(*args, **kwargs)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py", line 919, in writeto
    self.verify(option=output_verify)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/verify.py", line 73, in verify
    errs = self._verify(opt)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/hdu/hdulist.py", line 1245, in _verify
    result = hdu._verify(option)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/hdu/table.py", line 535, in _verify
    errs = super()._verify(option=option)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/hdu/base.py", line 1587, in _verify
    errs = super()._verify(option=option)
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/hdu/base.py", line 1074, in _verify
    errs.append(card._verify(option))
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/card.py", line 1118, in _verify
    keyword, valuecomment = self._split()
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/card.py", line 815, in _split
    for card in self._itersubcards():
  File "/home/skoposov/pyenv38/lib/python3.8/site-packages/astropy/io/fits/card.py", line 1170, in _itersubcards
    raise VerifyError(
astropy.io.fits.verify.VerifyError: Long card images must have CONTINUE cards after the first card.

```

### Steps to Reproduce

```python
import astropy.table as atpy
tab = atpy.Table().read('/tmp/00116M.track.eep', format='ascii')
tab.write('/tmp/xx.fits', overwrite=True)
```
The table can be taken here https://gist.github.com/segasai/7a65336b3420371f73a75a7dd2916f47

### System Details

Linux-5.8.0-50-generic-x86_64-with-glibc2.29
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0]
Numpy 1.20.2
astropy 4.2.2.dev4+g5bce144c6
Scipy 1.6.2
Matplotlib 3.4.1

Interestingly the code works in astropy -main branch, but fails with 4.2.x. 
I also know it used to work fine and only recently started to fail. 

The root cause of the problem seems to be related to the table.meta['comments'] which are very long and have all sorts of special symbols. It is clear that somewhere where those comments are added to the header, they are not sanitized enough. I'd expect them to fail here     https://github.com/astropy/astropy/blob/9d8014c822d993461ac2107030c308e4fc231e8c/astropy/io/fits/convenience.py#L588 
or have CONTINUE added. That doesn't seem to happen in 4.2.x branch. 
I couldn't quickly find why it works in the main branch to provide a fix.

**Comment by segasai** [comment]
Ah, after some more digging I saw the same issue  https://github.com/astropy/astropy/issues/11486 
with the patch that has been applied to master 
https://github.com/astropy/astropy/pull/11487
but not in v4.2.x branch.
I will keep this issue open just for the sake of 4.2. branch, but feel free to close.

**Comment by pllim** [comment]
If it is already fix on `main`, then you have to wait for 4.0.6, 4.2.2, or 4.3 to be released. Not sure if there is anything else we can do here. Thanks for the report and follow-ups!
